githubbot/githubbot/handler.go

```diff
@@ -67,57 +67,49 @@ func (h *Handler) HandleCommand(msg chat1.MsgSummary) error {
 
 	switch {
 	case strings.HasPrefix(cmd, "!github subscribe"):
-		h.handleSubscribe(cmd, msg, true, client)
+		return h.handleSubscribe(cmd, msg, true, client)
 	case strings.HasPrefix(cmd, "!github unsubscribe"):
-		h.handleSubscribe(cmd, msg, false, client)
+		return h.handleSubscribe(cmd, msg, false, client)
 	case strings.HasPrefix(cmd, "!github watch"):
-		h.handleWatch(cmd, msg.ConvID, true, client)
+		return h.handleWatch(cmd, msg.ConvID, true, client)
 	case strings.HasPrefix(cmd, "!github unwatch"):
-		h.handleWatch(cmd, msg.ConvID, false, client)
+		return h.handleWatch(cmd, msg.ConvID, false, client)
 	default:
 		h.Debug("ignoring unknown command")
 	}
 	return nil
 }
 
-func (h *Handler) handleSubscribe(cmd string, msg chat1.MsgSummary, create bool, client *github.Client) {
+func (h *Handler) handleSubscribe(cmd string, msg chat1.MsgSummary, create bool, client *github.Client) (err error) {
 	toks, err := shellquote.Split(cmd)
 	if err != nil {
-		h.ChatDebug(msg.ConvID, "error splitting command: %s", err)
-		return
+		return fmt.Errorf("error splitting command: %s", err)
 	}
 	args := toks[2:]
 	if len(args) < 1 {
-		h.ChatDebug(msg.ConvID, "bad args for subscribe: %s", args)
-		return
+		return fmt.Errorf("bad args for subscribe: %v", args)
 	}
 
 	var message string
 	defer func() {
 		if message != "" {
 			_, err = h.kbc.SendMessageByConvID(msg.ConvID, fmt.Sprintf(message, args[0]))
-			if err != nil {
-				h.ChatDebug(msg.ConvID, "Error sending message: %s", err)
-				return
-			}
+			err = fmt.Errorf("error sending message: %s", err)
```

Comment:
You only want to assign this if `err != nil`, otherwise we will always respond with this error message

Reply:
Oh good catch 🤦‍♂